### DISCUSSION TITLE
general curiosity - custom resources vs snowflake.core python APIs

### DISCUSSION THREAD

**Discussion by ncentola** [Titan-Systems/titan#161] [Q&A] [answered]
I currently manage a complex snowflake environment with pulumi and wanted to start exploring alternatives because pulumi is getting to be a burden. After poking around I stumbled upon this repo and thought it was really cool. I was curious though why you decided to design your resources from the ground up rather than build on top of the snowflake core python APIs for resource management.

**Comment by teej** [comment, marked as answer]
Great question! There are two reasons.

1. The Python APIs didn't exist when I started. Early on, the team at Snowflake asked if I wanted to donate Titan Core to merge into their project, but I decided against it.

2. The Python APIs are too slow for batch operations. Titan Core can manage 10,000 objects in just a few minutes, the Python APIs don't handle this scale well.

There's also the issue of resource coverage. They have only released 30 or so resource types. By using SQL, I don't have to wait months/years for Snowflake to release Python API support.

**Reply by ncentola** [reply]
makes a lot of sense, thanks!
